### ISSUE TITLE
When calling 'pre('remove')'.

### ISSUE THREAD

**Issue by sujed** [bradtraversy/devcamper-api#11]
Hello again.
I was building a front-end for this API and then I've ran into this problem.
When you remove the last course from the database, the ```pre('remove')``` hook counts the ```averageCost``` before the removal of the course, hence after it ran, the ```averageCost``` stays the same even tough we don't have any courses left in the database.
This is how i fixed the problem
![getAverageCost](https://user-images.githubusercontent.com/22971442/75567360-b0021a00-5a51-11ea-95cc-60da9ca8253d.png)
Basically if there is no courses left in the database then we should reset the ```averageCost``` to 0 or to undefined(so the field gets removed from the model).
Also I've made that the ```pre('remove')``` is now ```post('remove')``` because it makes sense to me to check the ```averageCost``` after the deletion was successful, and now our database is updated, and everything still works great!
Hope this solution helps and this wasn't too long! 
:)

**Comment by alfheimrShiven** [comment]
Hi @sujed, thanks for putting this forward. Modifying `pre('remove')` to `post('remove') `makes sense, and I feel that's the **only move required** to update the right `averageCost`

According to me, checking `if(obj.length>0)` isn't required, because if there are no courses linked to a particular `bootcampId`, the aggregate function will automatically result in `obj[0].averageCost=0` and the same will be updated.

**Comment by sujed** [comment]
Hello @alfheimrShiven the `if(obj.length>0)` is for the case when we delete the last course, I also thought that if there is no courses, it would do the average 0, but it didn't. If you leave it like this:
![bradSol1](https://user-images.githubusercontent.com/22971442/85921333-e9218e00-b87b-11ea-94e6-6fbafd9997fc.png)
You will get the 'cannot read property x of undefined' error because the obj would be []
so we have to check if there is any courses left.

**Comment by alfheimrShiven** [comment]
Thanks for letting me know the same @sujed. Appreciated!
Also, I noticed #25 pull request has been submitted addressing the same issue pointed out by you. Check it out!

**Comment by devrinku** [comment]
try this ,this works fine

if (obj[0]) {
      await this.model("Bootcamp").findByIdAndUpdate(bootcampid, {
        averageCost: Math.ceil(obj[0].averageCost / 10) * 10,
      });
    } else {
      await this.model("Bootcamp").findByIdAndUpdate(bootcampid, {
        averageCost: undefined,
      });
    }
